### ISSUE TITLE
Hello, friends, has the project been discontinued?

### ISSUE THREAD

**Issue by gomes2191** [gitahead/gitahead#598]
Hello, friends, the project was discontinued, there was never an update nor merge with the updates.

**Comment by hackhaslam** [comment]
It is probable that I will have very little time to work on GitAhead for the foreseeable future. I plan to make a release with already merged changes, but after that I will probably only make minor updates as necessary. Further development should be done in a fork unless somebody can show me that they have the ability to take it over entirely. I will add a similar notice to the README within the next few days.

**Comment by gomes2191** [comment]
Certainly, there must be someone who already contributes to the project with time and willingness to maintain it.

**Comment by hackhaslam** [comment]
Time and willingness aren't enough. They also have to be able to test and make builds for Windows, Linux, and macOS. The domain name and the certificates used to sign the macOS and Windows builds belong to the company I work for. It's not a trivial matter to transfer all of that organizational logic over to somebody new. It's possible, but it would require somebody who's really committed.

I really think that the best way forward is to keep GitAhead going in low-level maintenance mode, and encourage contributors that want to continue active development to rally around a fork that can be rebranded as a different tool.

**Comment by gomes2191** [comment]
I understand, but it will work.

**Comment by safield** [comment]
Could you not give someone permissions on the git repo to manage pull requests and maintain the code base, while you just do the minimum amount of tasks that require ownership of the signing certificates (basically just the macOS and Windows builds)?

**Comment by Sebclem** [comment]
@murmele Are you interested to continue developing in your fork ? 
A have tested your fork and there is very promising fix / features.
And i will be happy to help on any fork or on the main repo.

GitAhead is a very cool app, it will be so sad to let it die ...

**Comment by Murmele** [comment]
@Sebclem at the moment I don't have any time to do. But i will fix bugs in my branch if needed, or making merge request if somewhere elese hosted. You can ask at Gitlab if it possible to host the project there again, setting up all the continuous intergration/ deployment.  You can ask here if there are any guys who would like to help you to set everything up. I can help you if you have any specific questions about the double tree feature or about the code in general and with all the features in my branch

**Comment by Philipp91** [comment]
> You can ask at Gitlab if it possible to host the project there again

Why Gitlab?

The nice thing about GitHub is that the GitHub actions are already figured out and the main repository is here too.

> The domain name and the certificates used to sign the macOS and Windows builds belong to the company I work for. It's not a trivial matter to transfer all of that organizational logic over to somebody new.

Right. It seems like flatpak distribution (https://github.com/gitahead/gitahead/issues/597) is already set up. We could start with only that distribution method in a fork, as it's nicely automated and relatively publicly owned (doesn't require certificates AFAIU?).

> This guy seems to be killing it on the commits, maybe this fork could be the one people get behind...
https://github.com/Murmele/gitahead

I agree that most of the interesting functionality is there.

> at the moment I don't have any time to do.

Would you still be fine if more people joined in on the development under that fork (fixing up the GitHub actions like in https://github.com/gitahead/gitahead/pull/582, setting up a `gitahead-fork` flatpak, etc.)?

**Comment by advaithm** [comment]
Ping me if a fork arises. it should be easy enough to get gitahead marked as end-of-life if @hackhaslam  would like that.

**Comment by Murmele** [comment]
> > You can ask at Gitlab if it possible to host the project there again
> 
> Why Gitlab?
> 
> The nice thing about GitHub is that the GitHub actions are already figured out and the main repository is here too.
> 
> > The domain name and the certificates used to sign the macOS and Windows builds belong to the company I work for. It's not a trivial matter to transfer all of that organizational logic over to somebody new.
> 
> Right. It seems like flatpak distribution (#597) is already set up. We could start with only that distribution method in a fork, as it's nicely automated and relatively publicly owned (doesn't require certificates AFAIU?).
> 
> > This guy seems to be killing it on the commits, maybe this fork could be the one people get behind...
> > https://github.com/Murmele/gitahead
> 
> I agree that most of the interesting functionality is there.
> 
> > at the moment I don't have any time to do.
> 
> Would you still be fine if more people joined in on the development under that fork (fixing up the GitHub actions like in #582, setting up a `gitahead-fork` flatpak, etc.)?

I think you can start it also on github and I think here are a lot of people which would contribute including me at least for bug fixes or new ideas.

**Comment by Philipp91** [comment]
> I think you can start it also on github

By "you" did you mean me? Would you prefer the development to continue under my (empty) fork rather than yours (which already has much of the good stuff)?

> I think here are a lot of people which would contribute including me at least for bug fixes or new ideas.

So would you be willing to send the changes you made (at least the undoubtedly good ones that others also want) as separate, reviewable PRs?

**Comment by Murmele** [comment]
@Philipp91 I think the best would be to request at Github for a new Project and then hosting a fork there. Could be also my branch. Otherwise you are not able to handle PRs, issues, continuous integration and so on.

**Comment by Philipp91** [comment]
@advaithm How does the signing work for Flatpak? I guess we want some mechanism to ensure that no single actor/developer can push code (let alone code that's not from the public master branch or a public release). Is there a way in which Flatpak allows for that?

**Comment by Philipp91** [comment]
@Murmele E.g. [here](https://github.com/Murmele/gitahead/blob/826b4cf82591f078b6b7129ffa9691387bbee8f7/src/ui/DoubleTreeWidget.cpp#L180) with [`idClicked`](https://doc.qt.io/qt-5/qbuttongroup.html#idClicked) your fork seems to depend on Qt library functions that are available only from version 5.15, but the readme says it only requires 5.9. My naive Mint installation comes with 5.12, so I can't compile your fork.

> Otherwise you are not able to handle PRs, issues, continuous integration and so on.

I see that your fork already has an open PR (in fact it's from me :smiley:) , but no issue tracker (or I would have reported the above issue there). Same for my fork, but I can easily activate the issue tracker in the settings.

That said, I'm not opposed to creating a new GitHub project -- would `gitahead-fork` be a good name?

**Comment by advaithm** [comment]
if you are suggesting ci release, there are ways to do it but it would be safer to setup [flatpak-external-data-checker](https://github.com/flathub/flatpak-external-data-checker/) and have it update and rebuild when a new release is made

**Comment by advaithm** [comment]
> @Murmele E.g. [here](https://github.com/Murmele/gitahead/blob/826b4cf82591f078b6b7129ffa9691387bbee8f7/src/ui/DoubleTreeWidget.cpp#L180) with [`idClicked`](https://doc.qt.io/qt-5/qbuttongroup.html#idClicked) your fork seems to depend on Qt library functions that are available only from version 5.15, but the readme says it only requires 5.9. My naive Mint installation comes with 5.12, so I can't compile your fork.
> 
> > Otherwise you are not able to handle PRs, issues, continuous integration and so on.
> 
> I see that your fork already has an open PR (in fact it's from me :smiley:) , but no issue tracker (or I would have reported the above issue there). Same for my fork, but I can easily activate the issue tracker in the settings.
> 
> That said, I'm not opposed to creating a new GitHub project -- would `gitahead-fork` be a good name?

We would need to remove the name gitahead from our fork

**Comment by Murmele** [comment]
> @Murmele E.g. [here](https://github.com/Murmele/gitahead/blob/826b4cf82591f078b6b7129ffa9691387bbee8f7/src/ui/DoubleTreeWidget.cpp#L180) with [`idClicked`](https://doc.qt.io/qt-5/qbuttongroup.html#idClicked) your fork seems to depend on Qt library functions that are available only from version 5.15, but the readme says it only requires 5.9. My naive Mint installation comes with 5.12, so I can't compile your fork.
> 
> > Otherwise you are not able to handle PRs, issues, continuous integration and so on.
> 
> I see that your fork already has an open PR (in fact it's from me 😃) , but no issue tracker (or I would have reported the above issue there). Same for my fork, but I can easily activate the issue tracker in the settings.
> 
> That said, I'm not opposed to creating a new GitHub project -- would `gitahead-fork` be a good name?

Ah I didn't know that it is only a setting to get issuetracker and so on. I will check your pullrequest if I find it.
The problem is I'm using arch and therefore I have always the newest version. I think for users with older version flatpak is good or not?

> We would need to remove the name gitahead from our fork

I don't know the copyright on the name. But actually I like the name and especially the logo is pretty cool. Another option would be that @hackhaslam gives someone access to manage the project and so the project can continue.

**Comment by advaithm** [comment]
ideally I think getting a stable release along with flatpak for the release will solve a lot of issues. it would also allow us to support native wayland on linux

**Comment by fcore117** [comment]
This program is/was way better that those other heavily bloated git gui clients.

**Comment by domportera** [comment]
> This program is/was way better that those other heavily bloated git gui clients.

for real, I love it. Only problem is the latest release for Windows is super buggy. I'd eat my shoe for someone to compile a more recent build

**Comment by JulianGro** [comment]
Wouldn't it be the easiest to just give the top contributors permissions for this repository for now?
Moving repositories is always extremely annoying because you lose all the issues an PRs.

**Comment by JulianGro** [comment]
I would also be up for helping with maintenance by the way. I use GitAhead every day, but I also notice almost every day how buggy it is.

**Comment by Murmele** [comment]
Is there anybody who is familar with continuous deployment? I got in my fork CI running and the linux build is fine. MacOS and Windows does not. Maybe someone is able to fix the bug there and able to set up Continuous deployment

**Comment by Philipp91** [comment]
The Windows build complains at:

```
[520/589] Automatic MOC for target ui
...
[522/589] Building CXX object src\ui\CMakeFiles\ui.dir\ui_autogen\mocs_compilation.cpp.obj
FAILED: src/ui/CMakeFiles/ui.dir/ui_autogen/mocs_compilation.cpp.obj 
```

The Linux build doesn't even mention mocs_compilation anywhere and it only has `Automatic MOC for target tools`, but not `... for target ui`. Maybe that's why the Linux build doesn't run into this issue. And I don't know what MOC is and whether it should be running or not.

**Comment by advaithm** [comment]
MOC is a meta object compiler, it is required for gitahead. Ideally we should work on moving gitahead from a bundled qt to a system install of qt. https://doc.qt.io/archives/qt-4.8/moc.html

**Comment by domportera** [comment]
still watching this 👀

**Comment by hackhaslam** [comment]
I have finally made the change that I had previously proposed to both the README and the main web site.

**Comment by Murmele** [comment]
@hackhaslam thank you for your amazing work! Do you mean with rebranding the Icon and the name?

**Comment by hackhaslam** [comment]
@Murmele yes. It should just be clear that it's a different tool not maintained or endorsed by SciTools. Although, I suppose that it still should reproduce the SciTools copyright somewhere to conform with the license. I'm not a lawyer, but I believe that's essentially how it should work.

**Comment by stingray21** [comment]
I apologize if I overlooked or missed something, but is there a "a *rebranded* fork" that still is under active development? I love this tool and would like to keep using it. There are just some small features missing. I'm also willing to contribute, but I have only very basic C/C++ skills

**Comment by hackhaslam** [comment]
@stingray21 as far as I know, there isn't one yet.

**Comment by Murmele** [comment]
Is there anybody who has the knowledge what has to be done to create a fork?

**Comment by JulianGro** [comment]
Realistically you just press fork on the top right of the GitHub page. Then if they say it has to be rebranded, that would be the first order of buisness.
I don't think you can create a fork that keeps the open issues and PRs, but realistically most of those will be outdated anyways.
If you are up for being the lead maintainer, just press fork, change the settings to allow issues and PR requests (if they aren't on by standard), maybe create a Matrix channel or something so we can easily talk about things, and then just post here so we can join you.

**Comment by gomes2191** [comment]
Is there no one with technical competence capable of continuing the project?

**Comment by Murmele** [comment]
I set on my fork everything for developing. So we have to change the icons and the name, any proposals? I created a Matrix Chatroom: #GitAheadFork:matrix.org
But at the moment I don't have that much time for developing. I will merge pullrequests.

**Comment by stingray21** [comment]
@Murmele Are there any special settings to post in the Matrix chatroom? I get "Some of your messages have not been sent" (I can post in other chatrooms)

**Comment by JulianGro** [comment]
I can post, but I don't see any old messages (and no one seems to have typed something yet.
Also the channel is encrypted which doesn't really make too much sense, which is why I have never seen any other encrypted channel.

**Comment by Murmele** [comment]
now there are messages in the chat room

**Comment by Murmele** [comment]
> ```shell
> # https://github.com/Murmele/Gittyup
> wget -c https://github.com/Murmele/Gittyup/releases/download/latest/Gittyup.flatpak -O /tmp/Gittyup.flatpak && sudo flatpak install /tmp/Gittyup.flatpak
> ```

There is also a stable version on flathub: https://flathub.org/apps/details/com.github.Murmele.Gittyup
